NEWS.md

```diff
@@ -58,6 +58,8 @@
 
 7. `melt` returns an integer column for `variable` when `measure.vars` is a list of length=1, consistent with the documented behavior, [#5209](https://github.com/Rdatatable/data.table/issues/5209). Thanks to @tdhock for reporting and fixing. Any users who were relying on this behavior can change `measure.vars=list("col_name")` (output `variable` was column name, now is column index/integer) to `measure.vars="col_name"` (`variable` still is column name).
 
+8. `DT[,j,by]` could lose `class` attributes of `by` when `j` was GForce optimized, [#5567](https://github.com/Rdatatable/data.table/issues/5567). Thanks to @danwwilson for the report, and @ben-schwen for the PR.
```

Comment:
is this only the `class` attribute? e.g. what about `tzone` when grouping by POSIXt?

(only relevant to this issue/NEWS item if there's some inconsistency for `GForce` vs. non, please ignore / file for follow-up otherwise)

Reply:
the code comment mentions "attributes" generically, so I suspect that holds. we should add a test to this effect if so.

Reply:
Added a test to check the retention of generic attributes (`GForce` and non are now on par)